File: backend/app/models/subject.py
from sqlalchemy import Column, Integer, String, Boolean
from app.database import Base

class Subject(Base):
    __tablename__ = "subjects"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    code = Column(String(20), unique=True)
    weekly_lectures = Column(Integer, nullable=False)
    is_lab = Column(Boolean, default=False)
    lab_duration = Column(Integer, default=1)

File: SmartTimetable/src/SmartTimetable.Core/Entities/Subject.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTimetable.Core.Entities;

[Table("subjects")]
public class Subject
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [MaxLength(20)]
    [Column("code")]
    public string? Code { get; set; }

    [Required]
    [Column("weekly_lectures")]
    public int WeeklyLectures { get; set; }

    [Column("is_lab")]
    public bool? IsLab { get; set; }

    [Column("lab_duration")]
    public int? LabDuration { get; set; }
}
